cmake/scripts/linux/platform/X11.cmake

```diff
@@ -0,0 +1,8 @@
+core_require_dep(OpenGl)
+core_require_dep(EGL)
+core_require_dep(X)
+core_require_dep(XRandR)
+core_require_dep(LibDRM)
+core_optional_dep(VAAPI)
+core_optional_dep(VDPAU)
+core_optional_dep(OpenGLES)
```

Comment:
core_optional/require_dep would just have to loop over the parameters. Maybe that would allow us in the main cmake file to not use extra vars (optional/required_deps), but to directly use the functions. Could be done independently though.